cpu/arm7_common/arm_cpu.c

```diff
@@ -94,3 +94,47 @@ void thread_print_stack(void)
 
     printf("STACK (%d)= %X \n", i, *s);
 }
+
+void *thread_isr_stack_start(void)
+{
+    extern uintptr_t __stack_irq_start;
+    return (void *)&__stack_irq_start;
+}
+
+void *thread_isr_stack_pointer(void)
+{
+    unsigned _cpsr;
+    void *_sp;
+
+    /* get current mode */
+    __asm volatile ("mrs %0, cpsr" : "=r" (_cpsr));
+
+    /* If we are not in interrupt mode, the interrupt stack pointer will
+     * always be at the beginning of the interrupt stack.
+     */
+    if (_cpsr != 0x12) {
```

Comment:
You could just use `irq_is_in()` here. (Feel free to squash this in right away.)